evproc/processor.py

```diff
@@ -308,11 +310,19 @@ def process(self, ev):
                   ``StopProcessing`` exception raised by a processor.
         """
 
-        # Walk through all the processors and process the event
-        for proc in self._get_procs(ev):
-            try:
-                proc(ev)
-            except exc.StopProcessing as ex:
-                return ex.retval
+        event_deque = deque()
+        event_deque.append(ev)
+        while len(event_deque):
+            ev = event_deque.popleft()
+            # Walk through all the processors and process the event
+            for proc in self._get_procs(ev):
+                try:
+                    proc_result = proc(ev)
+                except exc.StopProcessing as ex:
+                    return ex.retval
+                if isinstance(proc_result, Event):
+                    event_deque.append(proc_result)
+                elif proc_result is not None:
```

Comment:
If you're importing collections, you could use isinstance() with collections.Sequence.  That would be true for strings as well, though, so you have to deal with that.